server/tests/src/main/scala/sttp/tapir/server/tests/ServerCancellationTests.scala

```diff
@@ -0,0 +1,55 @@
+package sttp.tapir.server.tests
+
+import cats.effect.IO
+import cats.effect.kernel.Async
+import cats.effect.syntax.all._
+import cats.syntax.all._
+import org.scalatest.matchers.should.Matchers._
+import sttp.client3._
+import sttp.monad.MonadError
+import sttp.tapir._
+import sttp.tapir.tests._
+
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.{Semaphore, TimeUnit}
+import scala.concurrent.duration._
+
+class ServerCancellationTests[F[_], OPTIONS, ROUTE](createServerTest: CreateServerTest[F, Any, OPTIONS, ROUTE])(implicit
```

Comment:
nice test :) I wonder which other backends support cancellation in this way ... but let's leave this for another task ;)

Reply:
also, this will be interesting to integrate into tapir-loom, but that's also another story